### PR TITLE
Update to napari plugin engine v2

### PR DESCRIPTION
**Pull request by dstansby**
This is work done by @noisysky to transfer to the new napari plugin engine - thanks! As far as I can tell using manual testing it works fine. Fixes https://github.com/brainglobe/brainglobe-napari-io/issues/6

### CODE REVIEW

**Review by alessandrofelder — COMMENTED** [review]
FWIW, changes to the code look as expected for the migration to me, and therefore reasonable.
Would still be good to test manually though (which I don't know how to), as Adam suggests.

**Comment by adamltyson** [comment]
Just tested it myself. All looks good, thanks @dstansby & @noisysky!

**Comment by adamltyson** [comment]
Whoops. I spoke too soon. The XML writer plugin doesn't seem to work. Tracked in https://github.com/brainglobe/brainglobe-napari-io/issues/10

**Comment by noisysky** [comment]
Yes, the writer never worked for me, even in the previous version. Sorry, didn't have time to look into it.

File: brainglobe_napari_io/napari.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    filename_patterns:
+    - '*.tiff'
+    accepts_directories: true
+  - command: brainglobe-napari-io.brainreg_read_dir_standard_space
+    filename_patterns:
+      - '*.tiff'
+    accepts_directories: true
+  - command: brainglobe-napari-io.cellfinder_read_dir
+    filename_patterns:
+    - '*.tif'
```
Comment (noisysky) [inline brainglobe_napari_io/napari.yaml:34]: Should probably be changed to accept both "tif" and "tiff"?

Comment (adamltyson) [reply brainglobe_napari_io/napari.yaml:34]: Where one is accepted, both should. I'm not yet familiar with `npe2`, this reader takes a directory as input, so do we need to specify a file type here?

**Pull request merged**